test/integration/testpaths/testpath_reduce_vm_deployments.py

```diff
@@ -0,0 +1,2073 @@
+
+        # Validate the following
+        # 1. Create ISO
+        # 2. Attach ISO to VM
+        # 3. Log in to the VM.
+        # 4. The device should be available for use
+        # 5. Detach ISO
+        # 6. Check the device is properly detached by logging into VM
+
+        if self.hypervisor.lower() in ["lxc"]:
+            self.skipTest("ISOs are not supported on LXC")
+
+        iso = Iso.create(
+            self.apiclient,
+            self.services["iso1"],
+            account=self.account.name,
+            domainid=self.account.domainid
+        )
+
+        self.debug("Successfully created ISO with ID: %s" % iso.id)
+        try:
+            iso.download(self.apiclient)
+        except Exception as e:
+            self.fail("Exception while downloading ISO %s: %s"
+                      % (iso.id, e))
+
+        self.debug("Attach ISO with ID: %s to VM ID: %s" % (
+            iso.id,
+            self.virtual_machine.id
+        ))
+        # Attach ISO to virtual machine
+        cmd = attachIso.attachIsoCmd()
+        cmd.id = iso.id
+        cmd.virtualmachineid = self.virtual_machine.id
+        self.apiclient.attachIso(cmd)
+
+        try:
+            ssh_client = self.virtual_machine.get_ssh_client()
+        except Exception as e:
+            self.fail("SSH failed for virtual machine: %s - %s" %
+                      (self.virtual_machine.ipaddress, e))
+
+        mount_dir = "/mnt/tmp"
+        cmds = "mkdir -p %s" % mount_dir
+        self.assert_(
+            ssh_client.execute(cmds) == [],
+            "mkdir failed within guest")
+
+        for diskdevice in self.services["diskdevice"]:
+            res = ssh_client.execute(
+                "mount -rt iso9660 {} {}".format(diskdevice, mount_dir))
+            if res == []:
+                self.services["mount"] = diskdevice
+                break
+        else:
+            self.fail("No mount points matched. Mount was unsuccessful")
+
+        c = "mount |grep %s|head -1" % self.services["mount"]
+        res = ssh_client.execute(c)
+        size = ssh_client.execute("du %s | tail -1" % self.services["mount"])
+        self.debug("Found a mount point at %s with size %s" % (res, size))
+
+        # Get ISO size
+        iso_response = Iso.list(
+            self.apiclient,
+            id=iso.id
+        )
+        self.assertEqual(
+            isinstance(iso_response, list),
+            True,
+            "Check list response returns a valid list"
+        )
+
+        try:
+            # Unmount ISO
+            command = "umount %s" % mount_dir
+            ssh_client.execute(command)
+        except Exception as e:
+            self.fail("SSH failed for virtual machine: %s - %s" %
+                      (self.virtual_machine.ipaddress, e))
+
+        # Detach from VM
+        cmd = detachIso.detachIsoCmd()
+        cmd.virtualmachineid = self.virtual_machine.id
+        self.apiclient.detachIso(cmd)
```

Comment:
again use library functions for detach iso .. Its always a good idea to no call api commands from test cases